compiler/rustc_parse/src/errors.rs

```diff
@@ -337,7 +337,7 @@ pub(crate) struct IfExpressionMissingThenBlock {
     #[primary_span]
     pub if_span: Span,
     #[subdiagnostic]
-    pub sub: IfExpressionMissingThenBlockSub,
+    pub sub: Vec<IfExpressionMissingThenBlockSub>,
```

Comment:
Instead of having a vector of subdiagnostics, could you make it something like:

```
pub sub: IfExpressionMissingThenBlockSub,
pub let_else_sub: Option<ExtraIfInLetElseSub>,
```

And then pass either `None` or `Some(ExtraIfInLetElseSub(span))` into the `IfExpressionMissingThenBlock` diagnostic?

That makes the code a bit easier to understand IMO.